app/src/main/java/com/therap/amin/currencyconverter/MainActivity.java

```diff
@@ -99,48 +86,24 @@ public boolean onOptionsItemSelected(MenuItem item) {
     }
 
     @Override
-    protected void onSaveInstanceState(Bundle outState) {
-        super.onSaveInstanceState(outState);
-        Log.d(TAG, "onSaveInstanceState: ");
-        outState.putString("output",output.getText().toString());
-        outState.putString("value",input.getText().toString());
-    }
-
-    @Override
-    protected void onRestoreInstanceState(Bundle savedInstanceState) {
-        super.onRestoreInstanceState(savedInstanceState);
-        input.setText(savedInstanceState.getString("value"));
-        output.setText(savedInstanceState.getString("output"));
-        Log.d(TAG, "onRestoreInstanceState: ");
-    }
-
-    @Override
     protected void onResume() {
         super.onResume();
 
-
-        Map<String, ?> allEntries = sharedPreferences.getAll();
-        if (allEntries.isEmpty()) {
-            Toast.makeText(getApplicationContext(),"Please set value at first",Toast.LENGTH_SHORT).show();
-            Intent intent = new Intent(MainActivity.this,CurrencyValueSaver.class);
+        boolean insideOnResumeForFirstTime = !sharedPreferences.contains(Constants.inputCurrencyKey) && onresumeExecution;
```

Comment:
logic complected. Rather use **startActivityForResult** method when calling **CurrencyValueSaverActivity**.  
And inside **CurrencyValueSaverActivity**, on **onBackPressed** method, use ``setResult(RESULT_OK, intent);`` method. See details on corresponding method documentation.